### PR TITLE
Copy and Paste Boundary Events

### LINKED ISSUE #1154: Allow Copying Boundary Events
Since detaching of boundary events is now possible it should be possible to copy boundary events. This is only prevented by the [`element.copy` rule](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/rules/BpmnRules.js#L902).

When disabling this rule:

foobar

### PR DESCRIPTION
**Pull request by oguzeroglu**
Closes #1154

Requires https://github.com/bpmn-io/diagram-js/pull/419 and https://github.com/bpmn-io/diagram-js/pull/418.

### CODE REVIEW

File: test/spec/features/modeling/behavior/DetachEventBehaviorSpec.js [hunk 1]
```diff
@@ -61,6 +62,29 @@ describe('features/modeling/behavior - detach events', function() {
       expect(boundaryEvent.type).to.equal('bpmn:BoundaryEvent');
       expect(boundaryEvent.businessObject.attachedToRef).to.equal(subProcess.businessObject);
     }));
+
+
+    it('should execute on pasting a BoundaryEvent outside', inject(function(canvas, elementRegistry, copyPaste) {
```
Comment (philippfromme) [inline test/spec/features/modeling/behavior/DetachEventBehaviorSpec.js]: This behavior is executed on `elements.create` (`modeling.createElements`), we don't want to test copy & paste here.

Also, please restructure the tests for readability:

```javascript
describe('basics', function() {
  describe('create elements', function() {
    it('should detach');
    it('should NOT detach');
  });
  describe('move elements', function() {
    it('should detach');
    it('should NOT detach');
  });
});
```

File: test/spec/features/copy-paste/BpmnCopyPasteSpec.js [hunk 2]
```diff
@@ -71,6 +71,75 @@ describe('features/copy-paste', function() {
...
+              x: task.x,
+              y: task.y
+            },
+            hints: {
+              attach: 'attach'
+            }
+          })[0].businessObject;
+
+          // then
+          expect(attachedBoundaryEvent.attachedToRef).to.eql(task.businessObject);
```
Comment (philippfromme) [inline test/spec/features/copy-paste/BpmnCopyPasteSpec.js]: Please keep the tests consistent. In the first test, you're checking the element's type, in the second you're checking the `attachedToRef` property.

```javascript
it('should copy and paste boundary event', function() {
  ...
  expect(businessObject.$type).to.equal('bpmn:IntermediateCatchEvent');
  expect(businessObject.attachedToRef).not.to.exist;
});

it('should copy and paste and reattach boundary event', function() {
  ...
  expect(businessObject.$type).to.equal('bpmn:BoundaryEvent');
  expect(businessObject.attachedToRef).to.exist;
});
```

**Comment by oguzeroglu** [comment]
@philippfromme Ready!

Also opened #1204 and #1205 for the issues came up while implementing this PR, will fix those as separate PRs.

**Comment by philippfromme** [comment]
Failing because both https://github.com/bpmn-io/diagram-js/pull/419 and https://github.com/bpmn-io/diagram-js/pull/418 are required.

File: test/spec/features/modeling/behavior/AttachEventBehaviorSpec.js [hunk 3]
```diff
@@ -5,171 +5,221 @@ import {
...
+        expect(intermediateEventBo.$type).to.equal('bpmn:IntermediateThrowEvent');
+        expect(intermediateEventBo.attachedToRef).not.to.exist;
+      }));
 
 
-  describe('event definition', function() {
+      it('should copy properties', inject(function() {
 
-    it('should copy event definitions', inject(function(elementRegistry, modeling) {
+        // TODO
```
Comment (nikku) [inline test/spec/features/modeling/behavior/AttachEventBehaviorSpec.js]: Please fill the gap. :cake:

**Comment by oguzeroglu** [comment]
@barmac this is fixed with https://github.com/bpmn-io/diagram-js/pull/413

**Comment by philippfromme** [comment]
> @barmac @philippfromme this is fixed with [bpmn-io/diagram-js#413](https://github.com/bpmn-io/diagram-js/pull/413) and #1208

We're on it. I'll fix it.

**Comment by philippfromme** [comment]
@barmac Can you check again? I can't reproduce this. Make sure you've linked diagram-js and checked out https://github.com/bpmn-io/diagram-js/tree/1154-allow-copying-boundary-events.

**Pull request merged**